From 7846188c587ca0ddc0c073f904a35d271bc0246c Mon Sep 17 00:00:00 2001
From: David Kershner <dkershner@habana.ai>
Date: Fri, 13 Jun 2025 05:11:14 +0300
Subject: [PATCH] [SW-226950] oobi: add sensor enables macros

Add several macros to support bit testing of the
SetStateSensorEnables composite sensor operational state.

Change-Id: I83c95ec1f2f652d065a546fb86ed341dd9706d3c
Signed-off-by: David Kershner <dkershner@habana.ai>
---

diff --git a/oobi.h b/oobi.h
index 93f0726..40ab115 100644
--- a/oobi.h
+++ b/oobi.h
@@ -43,6 +43,16 @@
 	OOBI_SENSOR_OP_STATE_UNAVAILABLE = 1,
 } oobi_sensor_op_state_t;
 
+#define HL_OOBI_PLDM_STATE_SENSOR_OP_STATE_UNAVAILABLE(index) \
+	BIT((index) * 2 + 1)
+#define HL_OOBI_PLDM_STATE_SENSOR_OP_STATE_IS_UNAVAILABLE(index, comp_op_state) \
+	(HL_OOBI_PLDM_STATE_SENSOR_OP_STATE_UNAVAILABLE(index) & (comp_op_state))
+
+#define HL_OOBI_PLDM_STATE_SENSOR_OP_STATE_DISABLED(index) \
+	BIT((index) * 2)
+#define HL_OOBI_PLDM_STATE_SENSOR_OP_STATE_IS_DISABLED(index, comp_op_state) \
+	(HL_OOBI_PLDM_STATE_SENSOR_OP_STATE_DISABLED(index) & (comp_op_state))
+
 struct hl_oobi_pldm_fu_component_state {
 	uint8_t state;
 	uint8_t pass_component_response_code;
